apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/UnauthenticatedAccessTest.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace Google.Cloud.Storage.V1.IntegrationTests
+{
+    [Collection(nameof(StorageFixture))]
+    public class UnauthenticatedAccessTest
+    {
+        // See https://cloud.google.com/storage/docs/public-datasets/landsat
+        // We pick one particular prefix/directory and a small file to test.
+        private const string LandsatBucket = "gcp-public-data-landsat";
+        private const string LandsatPrefix = "LC08/PRE/044/034/LC80440342016259LGN00/";
+        private const string LandsatObject = LandsatPrefix + "LC80440342016259LGN00_MTL.txt";
+
+        private readonly StorageFixture _fixture;
+
+        public UnauthenticatedAccessTest(StorageFixture fixture) => _fixture = fixture;
+
+        [Fact]
+        public void DownloadPublicData()
+        {
+            var client = StorageClient.CreateUnauthenticated();
```

Comment:
@jskeet by creating the client library this way will users be able to perform other operations if the appropriate IAM roles are given to "allUsers"? For example, upload a file if the bucket is writable for "allUsers" -- I hope no one uses a bucket this way, but wondering if it's possible.

Reply:
Users can *try* to do anything with such a client - whether they succeed will depend on the IAM configuration. I haven't tried creating a bucket that anyone can write to, but I *suspect* it would work fine. (That would certainly be consistent.)